libs/design/src/molecules/media-gallery/media-renderer/media-renderer.component.spec.ts

```diff
@@ -0,0 +1,80 @@
+import { Type } from '@angular/core';
+import {
+  ComponentFixture,
+  TestBed,
+  waitForAsync,
+} from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { BehaviorSubject } from 'rxjs';
+
+import { DaffArticleComponent } from '../../article/public_api';
+import { DaffCardComponent } from '../../card/public_api';
+import { DaffMediaGalleryComponent } from '../media-gallery.component';
+import { DaffMediaGalleryRegistry } from '../registry/media-gallery.registry';
+import { daffThumbnailCompatToken } from '../thumbnail/thumbnail-compat.token';
+import { DaffThumbnailDirective } from '../thumbnail/thumbnail.directive';
+import { DaffMediaRendererComponent } from './media-renderer.component';
+
+describe('DaffMediaRendererComponent', () => {
+  let component: DaffMediaRendererComponent;
+  let fixture: ComponentFixture<DaffMediaRendererComponent>;
+  let registry: DaffMediaGalleryRegistry;
+  let mockGallery: DaffMediaGalleryComponent;
+  let mockThumbnail1: DaffThumbnailDirective;
+  let mockThumbnail2: DaffThumbnailDirective;
+
+  beforeEach(waitForAsync(() => {
+    mockGallery = new DaffMediaGalleryComponent();
+    TestBed.configureTestingModule({
+      declarations: [
+        DaffMediaRendererComponent,
+        DaffArticleComponent,
+        DaffCardComponent,
+      ],
+      providers: [
+        {
+          provide: DaffMediaGalleryRegistry,
+          useValue: jasmine.createSpyObj('DaffMediaGalleryRegistry', ['add', 'remove', 'select']),
+        },
+        {
+          provide: DaffMediaGalleryComponent,
+          useValue: mockGallery,
+        },
+        { provide: daffThumbnailCompatToken, useExisting: DaffArticleComponent, multi: true },
```

Comment:
Instead of providing components that may change their dependencies in the future, can you instead just provide simple mock components?

Reply:
done